clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -349,19 +347,20 @@ private void sendProduceRequest(long now, int destination, short acks, int timeo
             recordsByPartition.put(tp, batch);
         }
 
-        ProduceRequest produceRequest = new ProduceRequest(acks, timeout, produceRecordsByPartition);
-        RequestHeader header = this.client.nextRequestHeader(ApiKeys.PRODUCE);
+        ProduceRequest.Builder requestBuilder =
+                new ProduceRequest.Builder(acks, timeout, produceRecordsByPartition);
```

Comment:
Hmm, how would the 0.9.0.1 broker even know that it was getting another request?  it doesn't know the size of the first request so it doesn't know where that request ends and a new one begins.